doc_source/aws_construct_lib.md

```diff
@@ -1,22 +1,63 @@
+### AWS CDK Stability Index<a name="aws_construct_lib_versioning_stability"></a>
 
-## CDK Patterns<a name="aws_construct_lib_patterns"></a>
+However, certain APIs deviate from the semantic versioning model\. The AWS CDK team has added a stability index to help developers determine which APIs deviate from the semantic versioning model\.
+
+There are three levels of stability in the AWS CDK Construct Library:
+
+Experimental  
+The API is still under active development and subject to non\-backward compatible changes or removal in any future version\. We recommend that you do not use this API in production environments\. Experimental APIs are not subject to the semantic versioning model\.
+
+Stable  
+The API is subject to the semantic versioning model\. We will not introduce non\-backward compatible changes or remove the API in a subsequent patch or feature release\.
+
+Deprecated  
+The API may emit warnings\. We do not guarantee backward compatibility\.
+
+Experimental and stable modules receive the same level of support from AWS\. The only difference is that we might change experimental APIs within a major version\. While we don not recommend using experimental APIs in production, we vet them the same way as we vet stable APIs before we include them in an AWS CDK release\.
+
+### Identifying the Support Level of an API<a name="aws_construct_lib_versioning_support"></a>
+
+The AWS CDK Developer guide and API Reference for each module starts with a section outlining the module's stability index\. The libraries that include only AWS CloudFormation Resources, and no hand\-curated constructs, are labelled with the maturity indicator **CloudFormation\-only**\.
+
+The module\-level gives an indication of the stability of the majority of the APIs included in the module, however individual APIs within the module can be annotated with different stability levels:
+
+
+| Stability | TypeScript | JavaScript | Python | C\#/\.NET | Java | 
+| --- |--- |--- |--- |--- |--- |
+| Experimental | @experimental | @stability Experimental | @experimental | Stability: Experimental | Stability: Experimental | 
+| Stable | @stable | @stability Stable | @stable | Stability: Stable | Stability: Stable | 
+| Deprecated | @deprecated | @Deprecated | @deprecated | \[Obsolete\] | Stability: Deprecated | 
+
+### Language Binding Stability<a name="aws_construct_lib_versioning_binding"></a>
+
+In addition to modules of the AWS CDK Construct Library, language support is also subject to a stability indication\. While the API described in all the languages is the same, the specific programming model and naming of language bindings considered experimental can change as it matures\.
+
+
+| Language | Stability | 
+| --- |--- |
+| TypeScript | Stable | 
+| JavaScript | Stable | 
+| Python | Stable | 
+| Java | Experimental | 
+| C\#/\.NET | Experimental | 
+
+## AWS CDK Patterns<a name="aws_construct_lib_patterns"></a>
```

Comment:
What's all this? Do not belong to the versioning section...

Reply:
It's from Romain's doc to indicate that Java and C#/.NET are also subject to non-major version breaking changes. Should I just remove the entire table/section?

Reply:
I was referring to everything below "## AWS CDK Patterns"

Reply:
It's because "Versioning" is in the "AWS Construct Library" topic and "AWS CDK Patterns" was already there. See https://docs.aws.amazon.com/cdk/latest/guide/aws_construct_lib.html. I'll remove it.

Reply:
Done. Please review all of the updates and let me know whether I need to make any other changes.